wandb/sdk/internal/internal_api.py

```diff
@@ -2472,12 +2472,16 @@ def upsert_sweep(
 
         config = self._validate_config_and_fill_distribution(config)
 
+        # Silly, but attr-dicts like EasyDicts don't serialize correctly to yaml.
+        # This sanitizes them with a round trip pass through json to get a regular dict.
+        config_str = yaml.dump(json.loads(json.dumps(config)))
```

Comment:
out of curiosity, do you have examples of what actually happens with those dict-like objects?

Reply:
also, do people use libs like hydra?

Reply:
oh yeah, it resultes in a fairly inscrutable error:
```
400 response executing GraphQL.
{"errors":[{"message":"Invalid sweep config: invalid hyperparameter configuration: dictitems","path":["upsertSweep"]}],"data":{"upsertSweep":null}}
wandb: ERROR Error while calling W&B API: Invalid sweep config: invalid hyperparameter configuration: dictitems (<Response [400]>)
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
/usr/local/lib/python3.9/dist-packages/wandb/sdk/lib/retry.py in __call__(self, *args, **kwargs)
    130             try:
--> 131                 result = self._call_fn(*args, **kwargs)
    132                 # Only print resolved attempts once every minute

20 frames
HTTPError: 400 Client Error: Bad Request for url: https://api.wandb.ai/graphql

During handling of the above exception, another exception occurred:

UsageError                                Traceback (most recent call last)
UsageError: Invalid sweep config: invalid hyperparameter configuration: dictitems

During handling of the above exception, another exception occurred:

CommError                                 Traceback (most recent call last)
/usr/local/lib/python3.9/dist-packages/wandb/util.py in no_retry_4xx(e)
   1005         return True
   1006     body = json.loads(e.response.content)
-> 1007     raise UsageError(body["errors"][0]["message"])
   1008 
   1009 

CommError: Invalid sweep config: invalid hyperparameter configuration: dictitems
```

Reply:
No idea about hydra, I'd never heard of it